### PR TITLE
sorter: extract methods and add tests

### PR DESCRIPTION
**Pull request by hi-rustin**
### What problem does this PR solve? 

None.

### What is changed and how it works?

trivial refactor for mem sorter.
1. extract methods
2. add tests

### Check List 

Tests 

 - Unit test

Code changes

None

Side effects

None

Related changes

None

### Release note 

```release-note
None
```

### CODE REVIEW

**Comment by hi-rustin** [comment]
/cc @overvenus @asddongmen 

Could you please take a look? Thanks!

**Review by amyangfei — APPROVED** [review]

File: cdc/sorter/memory/entry_sorter.go [hunk 1]
```diff
@@ -159,25 +130,57 @@ func (es *EntrySorter) Run(ctx context.Context) error {
...
 	}
-	es.lock.Unlock()
 }
 
 // Output returns the sorted raw kv output channel
 func (es *EntrySorter) Output() <-chan *model.PolymorphicEvent {
 	return es.outputCh
 }
 
+func eventLess(i *model.PolymorphicEvent, j *model.PolymorphicEvent) bool {
```
Comment (asddongmen) [inline cdc/sorter/memory/entry_sorter.go:152]: Is it better that use e1, e2 as args name instead of i, j?

**Review by ben1009 — APPROVED** [review]

File: cdc/sorter/memory/entry_sorter.go (hunk 1 above)
Comment (hi-rustin) [reply cdc/sorter/memory/entry_sorter.go:152]: i,j shows the index order, which is a bit more intuitive.

**Review by nongfushanquan — APPROVED** [review]

**Comment by hi-rustin** [comment]
/assign @amyangfei 

Could you please help trigger the merge process? Thanks!

**Pull request merged**